### PR TITLE
Backend Cleanup

### PR DESCRIPTION
**Pull request by tylerferrara**
DB
- changed date values from MM/DD/YYY to MM-DD-YYY to avoid escape characters

Backend
- Handle Pre-flight CORS requests with method OPTIONS (chrome sends two requests for header validation)
- Sanitized all incoming requests with regexp
- Send 404 on internal errors or edge cases that should not happen

Frontend
- made simple fetch call to test backend request

### CODE REVIEW

File: backend/main.go [hunk 1]
```diff
@@ -21,9 +22,18 @@ import (
...
+func sendResult(w http.ResponseWriter, v interface{}, success bool) {
+	var statCode = 404
+	if success {
+		statCode = 200
+	}
 	// enable CORS
 	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Access-Control-Allow-Credentials", "true")
+	w.Header().Set("Access-Control-Allow-Methods", "GET,HEAD,OPTIONS,POST,PUT")
```
Comment (sarahakbar16) [inline backend/main.go]: We only really use POST calls since we want to send a request body, so I'm not 100% sure if we need the other methods.

File: backend/main.go [hunk 2]
```diff
@@ -125,20 +161,20 @@ func registerHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	if ssn != respData.SSN || dob != respData.DOB {
 		pay.Message = "Not a citizen"
-		sendResult(w, pay)
+		sendResult(w, pay, true)
```
Comment (sarahakbar16) [inline backend/main.go:164]: If they're not a citizen, why are we returning a success? Is this success just to say "We talked to the backend successfully"?

File: backend/main.go [hunk 3]
```diff
@@ -125,20 +161,20 @@ func registerHandler(w http.ResponseWriter, r *http.Request) {
...
 		return
 	}
 	if isRegistered {
 		pay.Message = "Already registered"
-		sendResult(w, pay)
+		sendResult(w, pay, true)
```
Comment (sarahakbar16) [inline backend/main.go:169]: Same thing as above

File: frontend/src/Components/HomePage.js [hunk 4]
```diff
@@ -32,9 +32,25 @@ class HomePage extends React.Component {
             return this.setState({ error: 'Password is required' });
         }
 
-        return this.setState({ error: '' });
+        /* Send username and password to backend */
```
Comment (sarahakbar16) [inline frontend/src/Components/HomePage.js]: There's a file in the frontend that has the basic call to the backend (serviceCalls.js). I can change the call over there and add the method/headers/body changes in my PR so there's no merge conflict or anything.
